aptos-move/framework/aptos-stdlib/sources/string_utils.move

```diff
@@ -0,0 +1,143 @@
+module aptos_std::string_utils {
+    use std::string::String;
+
+    const EARGS_MISMATCH: u64 = 1;
+    const EINVALID_FORMAT: u64 = 2;
+
+    struct Cons<T, N> has copy, drop, store {
```

Comment:
nit: What do cons/car/cdr stand for? Can we use better names lol

Reply:
Those are the standard names

Reply:
They have been introduced in the 60ties for the ubiquitous lists in Lisp. But they have been a bit come out of fashion, and more frequently today, you find `head` and `tail`.

Reply:
are you a programming language barbarian? know thy classics

Reply:
Really, this is out of fashion since the 70ties, where functionally languages like Miranda, SML have their historic roots. Thats all you need to know. Anything without a decent type system can not be taken serious, is pre-classic, and in fact a bit barbarian.

Reply:
Sorry I meant the facitious response to kevin not knowing the names. I put the names in as wink to the old ages

Reply:
The only thing is that it's not a list in mirande/haskell sense as the second pair is not List<T> but N. Making it a generic pair with first/second kind of hides the intended list property we want.

Reply:
I'm wondering if there is more use general use of this then just formatting

Reply:
Lol I guess I'm not a computer scientist

Reply:
> The only thing is that it's not a list in mirande/haskell sense as the second pair is not List but N. Making it a generic pair with first/second kind of hides the intended list property we want.

LOL! Indeed it is totally untyped, a hack which allows us to workaround until we have either traits or C++ style overloading.

Reply:
totally untyped ==> totally dynamically typed